### ISSUE TITLE
Crash on startup

### ISSUE THREAD

**Issue by InsaanNotFound** [ChloeDawn/BeaconOverhaul#1]
```
[16:52:39] [main/FATAL]: Mixin apply failed mixins/beacons/mixins.server.json:BeaconBlockEntityMixin -> net.minecraft.class_2580: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException Critical injection failure: @Redirect annotation on updateBaseAndTier could not find any targets matching 'Lnet/minecraft/class_2580;method_16896(Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lnet/minecraft/class_2580;)V' in net.minecraft.class_2580. Using refmap mixins/beacons/refmap.json [PREINJECT Applicator Phase -> mixins/beacons/mixins.server.json:BeaconBlockEntityMixin -> Prepare Injections ->  -> redirect$zip000$updateBaseAndTier(Lnet/minecraft/class_1937;IIILnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lnet/minecraft/class_2580;)I -> Parse]
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Critical injection failure: @Redirect annotation on updateBaseAndTier could not find any targets matching 'Lnet/minecraft/class_2580;method_16896(Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lnet/minecraft/class_2580;)V' in net.minecraft.class_2580. Using refmap mixins/beacons/refmap.json [PREINJECT Applicator Phase -> mixins/beacons/mixins.server.json:BeaconBlockEntityMixin -> Prepare Injections ->  -> redirect$zip000$updateBaseAndTier(Lnet/minecraft/class_1937;IIILnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lnet/minecraft/class_2580;)I -> Parse]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.findMethods(InjectionInfo.java:572) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.readAnnotation(InjectionInfo.java:288) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.<init>(InjectionInfo.java:275) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.<init>(InjectionInfo.java:267) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.injection.struct.RedirectInjectionInfo.<init>(RedirectInjectionInfo.java:44) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_271]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_271]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_271]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_271]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo$InjectorEntry.create(InjectionInfo.java:140) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.parse(InjectionInfo.java:624) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.prepareInjections(MixinTargetContext.java:1247) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.prepareInjections(MixinApplicatorStandard.java:1033) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:388) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178) ~[sponge-mixin-0.8.2+build.24.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23) ~[fabric-loader-0.10.8.jar:0.8.2+build.24+unknown-b24.git-unknown]
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157) ~[fabric-loader-0.10.8.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150) ~[fabric-loader-0.10.8.jar:?]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_271]
	at net.minecraft.class_2246.<clinit>(class_2246.java:378) ~[intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.class_3523.<clinit>(class_3523.java:13) ~[intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.class_5471.<clinit>(class_5471.java:11) ~[intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.class_5458.method_30573(class_5458.java:44) ~[intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.class_5458.method_30566(class_5458.java:75) ~[intermediary-minecraft-1.16.4-client.jar:?]
	at java.util.LinkedHashMap.forEach(Unknown Source) [?:1.8.0_271]
	at net.minecraft.class_5458.<clinit>(class_5458.java:74) [intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.class_2378.<clinit>(class_2378.java:237) [intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.class_2966.method_12851(class_2966.java:42) [intermediary-minecraft-1.16.4-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:137) [intermediary-minecraft-1.16.4-client.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_271]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_271]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_271]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_271]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:224) [fabric-loader-0.10.8.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:141) [fabric-loader-0.10.8.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27) [fabric-loader-0.10.8.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_271]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_271]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_271]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_271]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Failed to start Minecraft:
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:141)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)
	... 8 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:224)
	... 10 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at net.minecraft.class_2246.<clinit>(class_2246.java:378)
	at net.minecraft.class_3523.<clinit>(class_3523.java:13)
	at net.minecraft.class_5471.<clinit>(class_5471.java:11)
	at net.minecraft.class_5458.method_30573(class_5458.java:44)
	at net.minecraft.class_5458.method_30566(class_5458.java:75)
	at java.util.LinkedHashMap.forEach(Unknown Source)
	at net.minecraft.class_5458.<clinit>(class_5458.java:74)
	at net.minecraft.class_2378.<clinit>(class_2378.java:237)
	at net.minecraft.class_2966.method_12851(class_2966.java:42)
	at net.minecraft.client.main.Main.main(Main.java:137)
	... 15 more
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [mixins/beacons/mixins.server.json:BeaconBlockEntityMixin] from phase [DEFAULT] in config [mixins/beacons/mixins.server.json] FAILED during APPLY
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:642)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:594)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:356)
	... 31 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Critical injection failure: @Redirect annotation on updateBaseAndTier could not find any targets matching 'Lnet/minecraft/class_2580;method_16896(Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lnet/minecraft/class_2580;)V' in net.minecraft.class_2580. Using refmap mixins/beacons/refmap.json [PREINJECT Applicator Phase -> mixins/beacons/mixins.server.json:BeaconBlockEntityMixin -> Prepare Injections ->  -> redirect$zip000$updateBaseAndTier(Lnet/minecraft/class_1937;IIILnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lnet/minecraft/class_2580;)I -> Parse]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.findMethods(InjectionInfo.java:572)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.readAnnotation(InjectionInfo.java:288)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.<init>(InjectionInfo.java:275)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.<init>(InjectionInfo.java:267)
	at org.spongepowered.asm.mixin.injection.struct.RedirectInjectionInfo.<init>(RedirectInjectionInfo.java:44)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo$InjectorEntry.create(InjectionInfo.java:140)
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.parse(InjectionInfo.java:624)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.prepareInjections(MixinTargetContext.java:1247)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.prepareInjections(MixinApplicatorStandard.java:1033)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:388)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 31 more
Exiting with -1
```

**Comment by DarianLStephens** [comment]
Looking at the sources, it appears this is targetting the Minecraft 1.17 snapshots, while you're trying to run it on 1.16.4.

**Comment by InsaanNotFound** [comment]
Funniest thing is that mod seemed to focus on 1.16, at least description said so

**Comment by ChloeDawn** [comment]
The [Reddit post I made](https://www.reddit.com/r/feedthebeast/comments/kgkaw5/beacons_an_overhaul_to_beacons_introducing_a_tier/) states that it targets 1.17. I forgot to include this in the repository README too, my apologies. I will be publishing on CurseForge very soon (waiting on a logo commission), and the release there will be under the new, less confusing name of "Beacon Overhaul", and will have a minimum version dependency of 20w49a.

**Comment by InsaanNotFound** [comment]
Ah good, thanks for explaining. Shame that it isn't on 1.16, good luck with development though! Closing.

**Comment by ChloeDawn** [comment]
@InsaanNotFound FYI, Beacon Overhaul is now available for 1.16

**Comment by InsaanNotFound** [comment]
Yeah, I tried it and I have bad news for ya


```
org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
    at Not Enough Crashes deobfuscated stack trace.(1.16.4+build.7)
    at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
    at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
    at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
    at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
    at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
    at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
    at java.lang.ClassLoader.loadClass(Unknown Source)
    at net.minecraft.client.MinecraftClient.render(MinecraftClient:1039)
    at net.minecraft.client.MinecraftClient.run(MinecraftClient:681)
    at net.minecraft.client.main.Main.main(Main:215)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
    at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:139)
    at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
    at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
    at org.multimc.EntryPoint.listen(EntryPoint.java:143)
    at org.multimc.EntryPoint.main(EntryPoint.java:34)
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector hasStatusEffectProxy(Lnet/minecraft/class_1309;Lnet/minecraft/class_1291;)Z in origins.mixins.json:BackgroundRendererMixin failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap Origins-1.16.4-refmap.json
    at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:408)
    at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1291)
    at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
    at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
    at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
    at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
    at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
    at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
    ... 24 more
```

Origins incompatible, causes game to be stuck on load and spam logs very fast

**Comment by ChloeDawn** [comment]
What is "Origins"?

**Comment by InsaanNotFound** [comment]
https://www.curseforge.com/minecraft/mc-mods/origins

**Comment by ChloeDawn** [comment]
Fixed in https://github.com/ChloeDawn/BeaconOverhaul/releases/tag/1.0.1-1.16
https://www.curseforge.com/minecraft/mc-mods/beaconoverhaul/files/3176388

(Also next time please make a separate issue ticket for new crashes)

**Comment by InsaanNotFound** [comment]
Noted
